verifiers.py

```diff
@@ -0,0 +1,121 @@
+    """
+    response = request(certs_url, method='GET')
+
+    if response.status != http_client.OK:
+        #TODO: Implement custom exceptions
+        raise exceptions.TransportError(
+            'Could not fetch certificates at {}'.format(certs_url))
+
+    return json.loads(response.data.decode('utf-8'))
+
+
+class Verifier(object):
+    """Verifies an ID token and returns the decoded token."""
+
+    def __init__(self, id_token, request, client_ids=None,
+            certs_url=_GOOGLE_OAUTH2_CERTS_URL):
+        """
+        Args:
+            id_token (Union[str, bytes]): The encoded token.
+            request (TODO: (Make new Request class) google.auth.transport.Request): The object used to make
+                HTTP requests.
+            client_ids (list[str]): List of CLIENT_ID values of all audiences that use this backend. If None,
+                then the audience is not verified.
+            certs_url (str): The URL that specifies the certificates to use to
+                verify the token. This URL should return JSON in the format of
+                ``{'key id': 'x509 certificate'}``.
+        """
+        self.id_token = id_token
+        self.request = request
+        self.clients = client_ids
+        self.certs_url = certs_url
+        self.verified_token = None
+        
+    def verify_token(self):
+        """
+        Verifies the stored ID Token and returns a DecodedToken object.
+
+        Returns:
+            DecodedToken: The decoded token.
+        """
+        if self.verified_token:
+            return self.verified_token
+        
+        certs = _fetch_certs(self.request, self.certs_url)
+
+        #TODO: implement JWT decode functionality
+        #self.verified_token = DecodedToken(self.id_token, certs, self.clients)
+        #return DecodedToken(self.id_token, certs, self.clients)
+        raise NotImplementedError("JWT DecodedToken needs to be implemented")
+        
+
+class Oauth2Verifier(Verifier):
+    """Verifies an ID Token issued by Google's OAuth 2.0 authorization server"""
+    
+    def __init__(self, id_token, request, client_ids=None):
+        """
+        Args:
+            id_token (Union[str, bytes]): The encoded token.
+            request (TODO: (Make new Request class)google.auth.transport.Request): The object used to make
+                HTTP requests.
+            client_ids (list[str]): List of CLIENT_ID values of all audiences that use this backend. If None,
+                then the audience is not verified.
+            certs_url (str): The URL that specifies the certificates to use to
+                verify the token. This URL should return JSON in the format of
+                ``{'key id': 'x509 certificate'}``.
+        """
+        Verifier.__init__(id_token, request, client_ids,
+                certs_url=_GOOGLE_OAUTH2_CERTS_URL) #uses OAuth2 url
+        
+
+class FirebaseVerifier(Verifier):
+    """Verifies an ID Token issued by Firebase Authentication"""
```

Comment:
I included this functionality because it existed in the original token library, but I see your point.  I can double check whether or not this is ever used in the context of Sign-In, and if it's not then I'll remove it.

Reply:
I appears that there's an existing firebase_admin library that contains methods for verifying id tokens through firebase apps, so that functionality isn't needed here.  I'll remove the FirebaseVerifier class from this repo.